### PR TITLE
Docs: Declaring dependency on eslint in shared config (fixes #6617)

### LINKED ISSUE #6617: eslint-config-standard@5.3.1 wants eslint@^2.0.0-rc.0
Just tried to install eslint for the first time following the readme and I got the following errors with `eslint --init` choosing the standard config.

**What version of ESLint are you using?**
3.0.1

**What parser (default, Babel-ESLint, etc.) are you using?**
default I assume?

**Please show your full configuration:**
```
{
    "extends": "standard",
    "plugins": [
        "standard"
    ]
}
```

**What did you do? Please include the actual source code causing the issue.**
There actually wasn't a 'npm-debug.log' file in directory in spite of what the error message indicated.

**What did you expect to happen?**
Successful --init

**What actually happened? Please include the actual, raw output from ESLint.**

```
$ npm install --save-dev eslint
eslint@3.0.1 node_modules/eslint
├── ignore@3.1.3
├── pluralize@1.2.1
├── path-is-inside@1.0.1
├── imurmurhash@0.1.4
├── estraverse@4.2.0
├── strip-bom@3.0.0
├── strip-json-comments@1.0.4
├── globals@9.9.0
├── esutils@2.0.2
├── progress@1.1.8
├── text-table@0.2.0
├── user-home@2.0.0 (os-homedir@1.0.1)
├── is-resolvable@1.0.0 (tryit@1.0.2)
├── debug@2.2.0 (ms@0.7.1)
├── doctrine@1.2.2 (esutils@1.1.6, isarray@1.0.0)
├── levn@0.3.0 (type-check@0.3.2, prelude-ls@1.1.2)
├── optionator@0.8.1 (fast-levenshtein@1.1.3, type-check@0.3.2, wordwrap@1.0.0, deep-is@0.1.3, prelude-ls@1.1.2)
├── json-stable-stringify@1.0.1 (jsonify@0.0.0)
├── chalk@1.1.3 (escape-string-regexp@1.0.5, supports-color@2.0.0, ansi-styles@2.2.1, has-ansi@2.0.0, strip-ansi@3.0.1)
├── require-uncached@1.0.2 (resolve-from@1.0.1, caller-path@0.1.0)
├── mkdirp@0.5.1 (minimist@0.0.8)
├── shelljs@0.6.0
├── concat-stream@1.5.1 (inherits@2.0.1, typedarray@0.0.6, readable-stream@2.0.6)
├── glob@7.0.5 (path-is-absolute@1.0.0, fs.realpath@1.0.0, inherits@2.0.1, once@1.3.3, inflight@1.0.5, minimatch@3.0.2)
├── espree@3.1.6 (acorn-jsx@3.0.1, acorn@3.2.0)
├── is-my-json-valid@2.13.1 (jsonpointer@2.0.0, generate-function@2.0.0, xtend@4.0.1, generate-object-property@1.2.0)
├── inquirer@0.12.0 (ansi-regex@2.0.0, strip-ansi@3.0.1, ansi-escapes@1.4.0, rx-lite@3.1.2, through@2.3.8, cli-width@2.1.0, figures@1.7.0, string-width@1.0.1, readline2@1.0.1, run-async@0.1.0, cli-cursor@1.0.2)
├── file-entry-cache@1.2.4 (object-assign@4.1.0, flat-cache@1.0.10)
├── js-yaml@3.6.1 (esprima@2.7.2, argparse@1.0.7)
├── table@3.7.8 (slice-ansi@0.0.4, tv4@1.2.7, xregexp@3.1.1, strip-ansi@3.0.1, string-width@1.0.1, bluebird@3.4.1)
├── es6-map@0.1.4 (d@0.1.1, es6-symbol@3.1.0, event-emitter@0.3.4, es6-iterator@2.0.0, es6-set@0.1.4, es5-ext@0.10.12)
├── escope@3.6.0 (esrecurse@4.1.0, es6-weak-map@2.0.1)
└── lodash@4.13.1
$ eslint --init
? How would you like to configure ESLint? Use a popular style guide
? Which style guide do you want to follow? Standard
? What format do you want your config file to be in? JSON
Installing eslint-plugin-standard, eslint-config-standard
npm WARN peerDependencies The peer dependency eslint@^2.0.0-rc.0 included from eslint-config-standard will no
npm WARN peerDependencies longer be automatically installed to fulfill the peerDependency 
npm WARN peerDependencies in npm 3+. Your application will need to depend on it explicitly.
npm WARN peerDependencies The peer dependency eslint-plugin-promise@^1.0.8 included from eslint-config-standard will no
npm WARN peerDependencies longer be automatically installed to fulfill the peerDependency 
npm WARN peerDependencies in npm 3+. Your application will need to depend on it explicitly.
npm ERR! Darwin 15.5.0
npm ERR! argv "/usr/local/bin/node" "/usr/local/bin/npm" "i" "--save-dev" "eslint-plugin-standard" "eslint-config-standard"
npm ERR! node v4.2.6
npm ERR! npm  v2.14.12
npm ERR! code EPEERINVALID

npm ERR! peerinvalid The package eslint@3.0.1 does not satisfy its siblings' peerDependencies requirements!
npm ERR! peerinvalid Peer eslint-config-standard@5.3.1 wants eslint@^2.0.0-rc.0

npm ERR! Please include the following file with any support request:
npm ERR!     /Users/rgant/Programming/.../npm-debug.log
Successfully created .eslintrc.json file in /Users/rgant/Programming/...
$ npm --version
2.14.12
```

### PR DESCRIPTION
**Pull request by alberto**

### CODE REVIEW

**Comment by platinumazure** [comment]
LGTM, thanks!

File: docs/developer-guide/shareable-configs.md [hunk 1]
```diff
@@ -26,6 +26,14 @@ Since `index.js` is just JavaScript, you can optionally read these settings for
 
 Once your shareable config is ready, you can [publish to npm](https://docs.npmjs.com/getting-started/publishing-npm-packages) to share with others. We recommend using the `eslint` and `eslintconfig` keywords so others can easily find your module.
 
+You should declare your dependency on eslint in `package.json` using the [peerDependecies](https://docs.npmjs.com/files/package.json#peerdependencies) field. The recommended way to declare a dependency for future proof compatibility is with the ">=" range syntax, using the lowest required eslint version. For example:
```
Comment (vitorbal) [inline docs/developer-guide/shareable-configs.md]: Small typo here: "peerDependecies" should be "peerDependencies"

File: docs/developer-guide/shareable-configs.md [hunk 2]
```diff
@@ -26,6 +26,14 @@ Since `index.js` is just JavaScript, you can optionally read these settings for
...
+
+```
+peerDependecies: {
```
Comment (vitorbal) [inline docs/developer-guide/shareable-configs.md]: Same on this line

**Comment by vitorbal** [comment]
LGTM, just a small typo!

**Comment by vitorbal** [comment]
Lgtm, thanks!

**Pull request merged**